test/agentchat/contrib/test_gpt_assistant.py

```diff
@@ -203,6 +206,35 @@ def test_get_assistant_files():
     assert expected_file_id in retrived_file_ids
 
 
+@pytest.mark.skipif(
+    sys.platform in ["darwin", "win32"] or skip_test,
+    reason="do not run on MacOS or windows or dependency is not installed",
+)
+def test_assistant_retrieval():
+    name = "For GPTAssistantAgent retrieval testing"
+
+    assistant_first = GPTAssistantAgent(
+        name,
+        instructions="This is a test",
+        llm_config={"config_list": config_list},
+    )
+    candidate_first = retrieve_assistants_by_name(assistant_first.openai_client, name)
+
+    assistant_second = GPTAssistantAgent(
+        name,
+        instructions="This is a test",
+        llm_config={"config_list": config_list},
+    )
+    candidate_second = retrieve_assistants_by_name(assistant_second.openai_client, name)
+
+    try:
+        assistant_first.delete_assistant()
+        assistant_second.delete_assistant()
+    except Exception:
```

Comment:
What exception is expected and why pass?

Reply:
If they are the same one assistant, deleting second assistant will be  be error on not found, in this case, it's expected, so we should pass it

Reply:
Got it. Can we catch the more specific error here? And add this comment.

Reply:
NP, make code friendly-reading is always better